Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java

```diff
@@ -500,7 +500,10 @@ public void addToList(Bundle rideRequestBundle) {
         try {
             Handler handler = new Handler(Looper.getMainLooper());
             handler.postDelayed(() -> executor.execute(() -> {
-                if (sheetArrayList.size() >= 3 || findCardById(rideRequestBundle.getString(getResources().getString(R.string.SEARCH_REQUEST_ID))))
+                String searchRequestId = rideRequestBundle.getString("searchRequestId");
+                if (sheetArrayList.size() >= 3 ||
+                        findCardById(searchRequestId) ||
+                        MyFirebaseMessagingService.clearedRideRequest.containsKey(searchRequestId))
```

Comment:
`MyFirebaseMessagingService.clearedRideRequest.containsKey(searchRequestId))`

In case where it exists in clearedRideRequest we have to skip and remove as well. Keeping requests for 30mins will just increase the `clearedRideRequest` size and memory